helix-term/src/commands.rs

```diff
@@ -5100,3 +5103,95 @@ fn rename_symbol(cx: &mut Context) {
     );
     cx.push_layer(Box::new(prompt));
 }
+
+/// If there is a number at the cursor, increment it by the count.
+fn increment_number(cx: &mut Context) {
+    increment_number_impl(cx, cx.count() as i64);
+}
+
+/// If there is a number at the cursor, decrement it by the count.
+fn decrement_number(cx: &mut Context) {
+    increment_number_impl(cx, -(cx.count() as i64));
+}
+
+/// If there is a number at the cursor, increment it by `amount`.
+fn increment_number_impl(cx: &mut Context, amount: i64) {
+    let (view, doc) = current!(cx.editor);
+
+    let selection = doc.selection(view.id);
+    if selection.len() != 1 {
+        return;
+    }
+
+    let primary_selection = selection.primary();
+    if primary_selection.to() - primary_selection.from() != 1 {
+        return;
+    }
+
+    let text = doc.text();
+    let cursor = primary_selection.cursor(text.slice(..));
+
+    if let Some(NumberInfo {
+        start,
+        end,
+        value: old_value,
+        radix,
+    }) = number_at(text, cursor)
```

Comment:
I think we should abstract this out so that it isn't just for number. The other thing I wish it is in vim is incrementing dates, there are too many dates but `YYYY-MM-DD` is good enough.

Reply:
I agree that the ability to increment dates would great, but maybe it could be added later in a separate PR?

Reply:
Yes, but still it is better to separate this number out.

Reply:
Another issue with incrementing dates is even if we limit it to `YYYY-MM-DD`, we don't know for sure it is a date. It could be just numbers being subtracted with no spaces between operations like this:
```python
num = 2021-11-10
```

Reply:
We can try and parse to see if it is a date but chances are it is a date rather than a number.